File: python/calculate_mode.py
def calculate_mode(input_list):
    """
    Given a list of integers or strings, return a list of the 
    most frequent values.
    """
    elt_dict = {}
    for elt in input_list:
        if elt in elt_dict:
            elt_dict[elt] +=1
        else:
            elt_dict[elt] = 1 #if not in dict, then add an entry with value 1
    mode = max(elt_dict.values()) #max value
    output = []
    for k,v in elt_dict.items():
        if v == mode:
            output.append(k)
    
    return sorted(output)

# print(calculate_mode([2,1,3]))


    

File: js/calculateMode.js
exports.calculateMode = (inputArray) => {
    eltObject = {}
    for (let elt of inputArray) {
        if (elt in eltObject) {
            eltObject[elt] ++ 
        } else {
            eltObject[elt] = 1
        } }
    
    const values = Object.values(eltObject)
    const mode = Math.max(...values)

    const output = []
    for (let[k,v] of Object.entries(eltObject)) {
        if (v === mode) {
            output.push(k)
        }}
    
    return output.sort()
}
// console.log(exports.calculateMode([4.5, 0, 0]))
